specification/search/data-plane/Azure.Search/preview/2024-05-01-Preview/searchindex.json

```diff
@@ -0,0 +1,3097 @@
+      "properties": {
+        "subscores": {
+          "$ref": "#/definitions/QueryResultDocumentSubscores",
+          "readOnly": true,
+          "description": "The breakdown of subscores of the document prior to the chosen result set fusion/combination method such as RRF."
+        }
+      }
+    },
+    "QueryResultDocumentSubscores": {
+      "type": "object",
+      "properties": {
+        "text": {
+          "type": "object",
+          "items": {
+            "$ref": "#/definitions/TextResult"
+          },
+          "readOnly": true,
+          "x-nullable": true,
+          "description": "The BM25 or Classic score for the text portion of the query."
+        },
+        "vectors": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/QueryResultDocumentVectorSubscores"
+          },
+          "readOnly": true,
+          "x-nullable": true,
+          "description": "The vector similarity and @search.score values for each vector query."
+        }
+      },
+      "description": "The breakdown of subscores between the text and vector query components of the search query for this document. Each vector query is shown as a separate object in the same order they were received."
+    },
+    "TextResult": {
+      "type": "object",
+      "properties": {
+        "searchScore": {
+          "type": "number",
+          "format": "double",
+          "readOnly": true,
+          "description": "The BM25 or Classic score for the text portion of the query."
+        }
+      },
+      "readOnly": true,
+      "additionalProperties": true,
+      "description": "The BM25 or Classic score for the text portion of the query."
+    },
+    "QueryResultDocumentVectorSubscores": {
+      "type": "object",
+      "additionalProperties": {
+        "type": "object",
+        "items": {
+          "$ref": "#/definitions/SingleVectorFieldResult"
+        },
+        "readOnly": true
+      },
+      "readOnly": true,
+      "description": "The vector similarity and @search.score values for each vector query. Each cross-field vector query will have separate subscores for each referenced field."
+    },
+    "SingleVectorFieldResult": {
+      "type": "object",
+      "properties": {
+        "searchScore": {
+          "type": "number",
+          "format": "double",
+          "readOnly": true,
+          "description": "The @search.score value that is calculated from the vector similarity score. This is the score that's visible in a pure single-field single-vector query."
+        },
+        "vectorSimilarity": {
+          "type": "number",
+          "format": "double",
+          "readOnly": true,
+          "description": "The vector similarity score for this document. Note this is the canonical definition of similarity metric, not the 'distance' version. For example, cosine similarity instead of cosine distance."
+        }
+      },
+      "readOnly": true,
+      "additionalProperties": true,
```

Comment:
You should avoid having additionalProperties in the same schema as named properties. Adding new properties then becomes a breaking change.